doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -424,50 +391,55 @@ automatically created.
          .. coqtop:: reset all
 
             Require Import List.
-            Hint Extern 5 ({?X1 = ?X2} + {?X1 <> ?X2}) => generalize  X1, X2; decide equality : eqdec.
+            Hint Extern 5 ({?X1 = ?X2} + {?X1 <> ?X2}) => generalize  X1, X2;
+              decide equality : eqdec.
             Goal forall a b:list (nat * nat), {a = b} + {a <> b}.
             Info 1 auto with eqdec.
 
-   .. cmdv:: Hint Cut @regexp : @ident
-      :name: Hint Cut
+   .. cmd:: Hint Cut [ @hints_regexp ] {? : {+ @ident } }
+
+      .. insertprodn hints_regexp hints_regexp
+
+      .. prodn::
+         hints_regexp ::= {+ @qualid }
+         | _
+         | @hints_regexp %| @hints_regexp
+         | @hints_regexp @hints_regexp
+         | @hints_regexp *
+         | emp
+         | eps
+         | ( @hints_regexp )
+
+      The `emp` regexp does not match any search path while `eps`
+      matches the empty path.
+
+      Used to cut the proof search tree according to a regular
+      expression that matches the paths to be cut. Note that there is no
+      operator precedence during parsing, one can
+      check with :cmd:`Print HintDb` to verify the current cut expression:
 
       .. warning::
 
          These hints currently only apply to typeclass proof search and the
          :tacn:`typeclasses eauto` tactic.
 
-      This command can be used to cut the proof-search tree according to a regular
-      expression matching paths to be cut. The grammar for regular expressions is
-      the following. Beware, there is no operator precedence during parsing, one can
-      check with :cmd:`Print HintDb` to verify the current cut expression:
+      .. todo PR: Should we keep/change 79 uses of nonbreaking space in the rsts?
```

Comment:
These were often introduced on purpose to fix what would otherwise be badly rendered. So no, except on a case by case basis.